src/service/performance-impl.js

```diff
@@ -617,6 +643,25 @@ export class Performance {
     }
   }
 
+  /**
+   * Record the interaction to next paint score.
+   * @param {number=} duration
+   */
+  tickInteractionToNextPaint_(duration) {
+    if (!this.ampdoc_) {
+      return;
+    }
+
+    const old = this.metrics_.get(TickLabel_Enum.INTERACTION_TO_NEXT_PAINT);
+    if (old == null || duration > old) {
+      this.tickDelta(
+        TickLabel_Enum.INTERACTION_TO_NEXT_PAINT,
+        duration - (old ?? 0)
+      );
+      this.flush();
```

Comment:
does this mean we just keep on reporting subsequent events that are higher than the old event?

Reply:
Yes.

Reply:
asking if this would cause a bias if we were computing the percentiles of INP

Reply:
that sounds like a good idea 👍

Reply:
do you plan to implement it in this PR or follow up?

Reply:
@erwinmombay It is the publisher's responsibility to configure their trigger. They can use [on:timer trigger](https://amp.dev/documentation/components/amp-analytics#%22on%22:-%22timer%22-trigger) to achieve this in their analytics config. We are providing flexibility by providing what is the max event duration since the page load till now (whichever it is).

Reply:
do we have any documentation around that? or is that already clear with the current timer trigger documentation?

Reply:
besides amp-analytics, what will this report to our csi (our own instrumentation)?

Reply:
It is similar to CLS  (which is also a continuous metrics rather than an event).

Reply:
following up here. i think we need to either implement a similar threshold or put this under an experiment flag for now until we iron out the details. i think with the current implementation this will cause a bias on the lower end of the distribution since we will be reporting more data points it

Reply:
for now lets go ahead and experiment flag guard this so we can iterate on the implementation and merge this

Reply:
friendly ping @powerivq , could we try and make any progress on this